lib/internal/modules/cjs/loader.js

```diff
@@ -435,7 +435,7 @@ function resolveBasePath(basePath, exts, isMain, trailingSlash, request) {
 
 function trySelf(parentPath, isMain, request) {
   const { data: pkg, path: basePath } = readPackageScope(parentPath) || {};
-  if (!pkg || 'exports' in pkg === false) return false;
+  if (!pkg || pkg.exports === undefined) return false;
```

Comment:
this still would pass if an inherited `exports` property existed.
```suggestion
  if (!pkg || !HasOwnProperty(pkg, 'exports')) return false;
```
(also bring in `HasOwnProperty` from `primordials` at the top)

Reply:
Isn't that similar to the first commit (https://github.com/nodejs/node/pull/31427/commits/d3f5674fe9fad844f9319a26bc850c7f62a1e6b3) which doesn't work (`pkg.exports` does exist but has an `undefined` value)?

Reply:
in that case you'd want both checks.